domain_alias/src/DomainAliasListBuilder.php

```diff
@@ -47,6 +47,11 @@ public function buildRow(EntityInterface $entity) {
    * {@inheritdoc}
    */
   public function render() {
+
+    $current_path = \Drupal::service('path.current')->getPath();
```

Comment:
And this filtering should happen at the getEntityIds() level, which can be overridden by this class.

I can't see immediately how to upcast the $domain from the route without rewriting this as a Controller. 

But you should not parse the url, use the notes here -- https://www.drupal.org/node/2310429